uint32 nNewDirection
uint32 nNewSpeed
---
uint32 nLastDirection
uint32 nLastSpeed
int32 nRetCode
